pkg/service/service.go

```diff
@@ -661,6 +674,10 @@ func filterServiceBackends(svc *svcInfo, onlyPorts []string) map[string][]*lb.Ba
 		}
 		// check for either named port
 		for _, backend := range svc.backends {
+			if hasPreferredBE && backend.Preferred == lb.Preferred(true) {
```

Comment:
If the comment above seems valid to you, then `hasPreferredBE` definitin may be moved to above this loop to make the variable definition and usage close to each other

Reply:
thanks, I have simplified the patch, and added the test cases for the same.